### DISCUSSION TITLE
ripgrep backend for searching commits

### DISCUSSION THREAD

**Discussion by sankalp-khare** [magit/magit#5113] [Ideas]
In large repositories when I filter logs by commit contents (`-G Search Changes`), it takes pretty long. Wondering if it would be viable (like is it even possible) to make the search tool configurable. I would then use something like `ripgrep`.

**Comment by tarsius** [comment]
Magit just calls `git log -G`; how the searching is done depends all on Git. Briefly looking at the relevant Git documentation again, I do not spot anything that would allow delegating the searching to an external tool.
